core/io/resource_importer.cpp

```diff
@@ -126,22 +129,45 @@ Error ResourceFormatImporter::_get_path_and_type(const String &p_path, PathAndTy
 
 Ref<Resource> ResourceFormatImporter::load(const String &p_path, const String &p_original_path, Error *r_error, bool p_use_sub_threads, float *r_progress, CacheMode p_cache_mode) {
 	PathAndType pat;
+	Ref<Resource> res;
+
+#ifdef TOOLS_ENABLED
+retry:
+	bool can_retry = false;
+	if (Engine::get_singleton()->is_editor_hint() && EditorFileSystem::get_singleton()->doing_first_scan()) {
+		can_retry = true;
+		// Disable error output to prevent displaying all the errors from accessing resource
+		// that are not yet loaded.
+		set_error_log_enabled(false);
+	}
+#endif
+
 	Error err = _get_path_and_type(p_path, pat);
 
 	if (err != OK) {
 		if (r_error) {
 			*r_error = err;
 		}
-
-		return Ref<Resource>();
+		res = Ref<Resource>();
+	} else {
+		res = ResourceLoader::_load(pat.path, p_path, pat.type, p_cache_mode, r_error, p_use_sub_threads, r_progress);
 	}
 
-	Ref<Resource> res = ResourceLoader::_load(pat.path, p_path, pat.type, p_cache_mode, r_error, p_use_sub_threads, r_progress);
-
 #ifdef TOOLS_ENABLED
 	if (res.is_valid()) {
 		res->set_import_last_modified_time(res->get_last_modified_time()); //pass this, if used
 		res->set_import_path(pat.path);
+	} else if (can_retry && EditorFileSystem::get_singleton()->doing_first_scan()) {
+		can_retry = false;
+		set_error_log_enabled(true);
+		err = EditorFileSystem::get_singleton()->import_file(p_path, false);
+		if (err == OK) {
+			goto retry;
```

Comment:
💀

Reply:
I guess you don't like the goto?? I saw that elsewhere in the base code so I used it, but I'll change it!

Reply:
Well `goto` is generally frowned upon. The Godot code that uses it is likely very old.

Reply:
Used a while instead.